optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/quad/AbstractQuadJoiner.java

```diff
@@ -20,14 +20,26 @@
 import java.util.List;
 import java.util.function.Function;
 
+import org.optaplanner.core.api.function.QuadPredicate;
 import org.optaplanner.core.api.function.TriFunction;
 import org.optaplanner.core.api.score.stream.quad.QuadJoiner;
 import org.optaplanner.core.impl.score.stream.common.AbstractJoiner;
+import org.optaplanner.core.impl.score.stream.common.JoinerType;
 
 public abstract class AbstractQuadJoiner<A, B, C, D> extends AbstractJoiner implements QuadJoiner<A, B, C, D> {
 
+    private final QuadPredicate<A, B, C, D> filter;
+
+    protected AbstractQuadJoiner() {
+        this.filter = null;
+    }
+
+    protected AbstractQuadJoiner(QuadPredicate<A, B, C, D> filter) {
+        this.filter = filter;
+    }
+
     @SafeVarargs
-    public final static <A, B, C, D> QuadJoiner<A, B, C, D> merge(QuadJoiner<A, B, C, D>... joiners) {
+    public static <A, B, C, D> AbstractQuadJoiner<A, B, C, D> merge(QuadJoiner<A, B, C, D>... joiners) {
```

Comment:
(soft) consistency: In AbstractTriJoiner this method signature is different: it's still final (what about Uni & Bi).